src/plugins/controls/public/react_controls/controls/data_controls/options_list_control/fetch_and_validate.tsx

```diff
@@ -57,6 +61,12 @@ export function fetchAndValidate$({
     stateManager.searchTechnique,
     // cannot use requestSize directly, because we need to be able to reset the size to the default without refetching
     api.loadMoreSubject.pipe(debounceTime(100)), // debounce load more so "loading" state briefly shows
+    apiPublishesReload(api.parentApi)
+      ? api.parentApi.reload$.pipe(
+          tap(() => pluginServices.getServices().optionsList.clearOptionsListCache()),
```

Comment:
We should be clearing the **new** `OptionsListFetchCache`, no? Not the old one in the services 🤔 You'll need to add a method to `options_list_fetch_cache.ts`

Reply:
Thanks for pointing this out. I did not realize these were different.